crates/modelardb_compression/src/merge.rs

```diff
@@ -58,141 +58,272 @@ pub fn try_merge_segments(compressed_segments: RecordBatch) -> Result<RecordBatc
+            } else {
+                // Merge timestamps and compute the new segment's minimum and maximum value.
+                let mut timestamp_builder = TimestampBuilder::new();
+
+                // NaN is used as the initial value as it will be overwritten be any value. Thus, it
+                // is the only initial value that preserves a minimum and maximum value that is NaN.
+                let mut min_value = f32::NAN;
+                let mut max_value = f32::NAN;
+
+                for maybe_segment_index in merge_indices
+                    .iter()
+                    .take(end_merge_index + 1)
+                    .skip(start_merge_index)
+                {
+                    // unwrap() is safe as the range of index into merge_indices are all Some.
+                    let segment_index = maybe_segment_index.unwrap();
+
+                    timestamps::decompress_all_timestamps(
+                        start_times.value(segment_index),
+                        end_times.value(segment_index),
+                        timestamps.value(segment_index),
+                        &mut timestamp_builder,
+                    );
+
+                    min_value = f32::min(min_value, min_values.value(segment_index));
+                    max_value = f32::max(max_value, max_values.value(segment_index));
+                }
+
+                let timestamps = timestamp_builder.finish();
+                let compressed_timestamps =
+                    timestamps::compress_residual_timestamps(timestamps.values());
+
+                // Merge the models.
+                // unwrap() is safe as the range of index into merge_indices are all Some.
+                let segment_index = merge_indices[end_merge_index].unwrap();
+                let model_type_id = model_type_ids.value(segment_index);
+                let values = values.value(segment_index);
+                let residuals = residuals.value(segment_index);
+                let values = models::merge(model_type_id, min_value, max_value, values);
+
+                // The metadata shared for all segment's are read from the last segment. In
+                // addition, since only the last segment may contain residuals, they are copied.
+                merged_compressed_segments.append_compressed_segment(
+                    univariate_ids.value(segment_index),
+                    model_type_ids.value(segment_index),
+                    timestamps.value(0),
+                    timestamps.value(timestamps.len() - 1),
+                    &compressed_timestamps,
+                    min_value,
+                    max_value,
+                    &values,
+                    residuals,
+                    errors.value(segment_index),
+                );
+
+                start_merge_index = end_merge_index + 2;
+            }
+        }
+    }
+
+    Ok(merged_compressed_segments.finish())
+}
+
 #[cfg(test)]
 mod tests {
+    use std::sync::Arc;
+
+    use crate::ErrorBound;
```

Comment:
Maybe move this down since we normally have `use crate` in the third block.

Reply:
Fixed.